File: iOS/Kalorie/Features/MealTypeSheet/MealTypeSheetRouter.swift
//
//  MealTypeSheetRouter.swift
//  Kalorie
//
//  Created by Josef Antoni on 19.09.2026.
//

import Foundation

struct MealTypeSheetRouter {

    // MARK: - Properties

    private let exportConfigurator: ExportConfigurator

    // MARK: - Init

    init(exportConfigurator: ExportConfigurator = ExportConfigurator()) {
        self.exportConfigurator = exportConfigurator
    }

    // MARK: - Functions

    func makeExportView(mealTypes: [MealTypeDomain]) -> ExportView {
        exportConfigurator.createView(mealTypes: mealTypes)
    }
}

File: Android/app/src/main/java/antoni/kalorie/features/mealtypesheet/MealTypeSheetRouter.kt
package antoni.kalorie.features.mealtypesheet

import androidx.compose.runtime.Composable
import antoni.kalorie.core.models.MealTypeDomain
import antoni.kalorie.features.export.ExportConfigurator

class MealTypeSheetRouter(
    private val exportConfigurator: ExportConfigurator = ExportConfigurator(),
) {

    // MARK: - Functions

    @Composable
    fun makeExportView(mealTypes: List<MealTypeDomain>, onBack: () -> Unit) {
        exportConfigurator.createView(mealTypes = mealTypes, onBack = onBack)
    }
}
